### PR TITLE
Refinements to Change official DOB journey

### PR DESCRIPTION
**Pull request by gunndabad**
- Initialize the date on the details page to what the user entered when coming from the Confirm page.
- Redirect back to the Confirm page when clicking Continue on the Details page when the user has come from the Confirm page.
- Use the name of the uploaded file on the Confirm page rather than our generated name.

There's a similar set of changes coming separately for the name page.

This approach of passing all data as query params (with a signature) may not be scalable long-term but I've left it as-is for now.

### CODE REVIEW

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Helpers/QueryStringSignatureHelper.cs [hunk 1]
```diff
@@ -31,7 +31,9 @@ public bool VerifySignature(Url url)
     private string CalculateSignature(Url url)
     {
         var canonicalUrl = new Url(url.Path + "?" + url.Query)
-            .RemoveQueryParam(SignatureParameterName);
+            .RemoveQueryParam(SignatureParameterName)
+            .ToString()
+            .TrimEnd('?');
```
Comment (gunndabad) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Helpers/QueryStringSignatureHelper.cs:36]: There was some inconsistency previously when the `url` had no query parameters; sometimes this URL had a `?` at the end and sometimes it didn't (depending on exactly how it was constructed, seemingly).

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Infrastructure/ModelBinding/DateOnlyModelBinder.cs [hunk 2]
```diff
@@ -30,11 +43,11 @@ public class DateOnlyModelBinderProvider : IModelBinderProvider
 {
     public IModelBinder? GetBinder(ModelBinderProviderContext context)
     {
-        if (context.Metadata.UnderlyingOrModelType == typeof(DateOnly) &&
-            (context.BindingInfo.BindingSource == BindingSource.Query ||
-            (context.BindingInfo.BindingSource == BindingSource.Path)))
+        if (context.Metadata.UnderlyingOrModelType == typeof(DateOnly))
```
Comment (gunndabad) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Infrastructure/ModelBinding/DateOnlyModelBinder.cs:46]: The removal of the check around `BindingSource` was to make the `[BindProperty(SupportsGet = true)] DateOnly DateOfBirth` property on the Details page work correctly with both query parameters and POST form submissions.

File: dotnet-authserver/src/TeacherIdentity.AuthServer/TeacherIdentity.AuthServer.csproj [hunk 3]
```diff
@@ -19,7 +19,7 @@
     <PackageReference Include="EFCore.NamingConventions" Version="7.0.2" />
     <PackageReference Include="FluentValidation.DependencyInjectionExtensions" Version="11.5.1" />
     <PackageReference Include="Flurl" Version="3.0.7" />
-    <PackageReference Include="GovUk.Frontend.AspNetCore" Version="1.0.0-beta4" />
+    <PackageReference Include="GovUk.Frontend.AspNetCore" Version="1.0.0-rc1" />
```
Comment (gunndabad) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/TeacherIdentity.AuthServer.csproj:22]: This version exposes the built-in `DateOnly` model binder so we can layer it with our own.

**Comment by gunndabad** [comment]
[Change DQT name/DOB snags](https://trello.com/c/4m1sqWQ7/918-change-dqt-name-dob-snags)

**Review by wilpow-github — APPROVED** [review]

**Pull request merged**